sim/warrior/fury/fury.go

```diff
@@ -39,28 +40,52 @@ func NewFuryWarrior(character *core.Character, options *proto.Player) *FuryWarri
 		Options: furyOptions,
 	}
 
-	// rbo := core.RageBarOptions{
-	// 	StartingRage:   furyOptions.ClassOptions.StartingRage,
-	// 	RageMultiplier: core.TernaryFloat64(war.Talents.EndlessRage, 1.25, 1),
-	// }
-	// if mh := war.GetMHWeapon(); mh != nil {
-	// 	rbo.MHSwingSpeed = mh.SwingSpeed
-	// }
-	// if oh := war.GetOHWeapon(); oh != nil {
-	// 	rbo.OHSwingSpeed = oh.SwingSpeed
-	// }
+	rbo := core.RageBarOptions{
+		StartingRage: furyOptions.ClassOptions.StartingRage,
+	}
+	if mh := war.GetMHWeapon(); mh != nil {
+		rbo.MHSwingSpeed = mh.SwingSpeed
+	}
+	if oh := war.GetOHWeapon(); oh != nil {
+		rbo.OHSwingSpeed = oh.SwingSpeed
+	}
 
-	// war.EnableRageBar(rbo)
-	// war.EnableAutoAttacks(war, core.AutoAttackOptions{
-	// 	MainHand:       war.WeaponFromMainHand(war.DefaultMeleeCritMultiplier()),
-	// 	OffHand:        war.WeaponFromOffHand(war.DefaultMeleeCritMultiplier()),
-	// 	AutoSwingMelee: true,
-	// 	ReplaceMHSwing: war.TryHSOrCleave,
-	// })
+	war.EnableRageBar(rbo)
+	war.EnableAutoAttacks(war, core.AutoAttackOptions{
+		MainHand:       war.WeaponFromMainHand(war.DefaultMeleeCritMultiplier()),
+		OffHand:        war.WeaponFromOffHand(war.DefaultMeleeCritMultiplier()),
+		AutoSwingMelee: true,
+	})
 
 	return war
 }
 
+func (war *FuryWarrior) RegisterSpecializationEffects() {
+
+	// Unshackled Fury
+	war.RegisterMastery()
+
+	// Dual Wield specialization
+	war.AutoAttacks.OHConfig().DamageMultiplier *= 1.25
+
+	// Precision
+	war.AddStat(stats.MeleeHit, 3*core.MeleeHitRatingPerHitChance)
+}
+
+func (war *FuryWarrior) GetMasteryBonusMultiplier() float64 {
+	return 1 + (11.2+5.6*war.GetMasteryPoints())/100
+}
+
+func (war *FuryWarrior) RegisterMastery() {
+	// The actual effects of Unshackled Fury need to be handled by specific spells
+	// as it modifies the "benefit" of them (e.g. it both increases Raging Blow's damage
+	// and Enrage's damage bonus) so we just keep this field up to date for spells to read
+	war.PseudoStats.EnragedEffectMultiplier = war.GetMasteryBonusMultiplier()
+	war.AddOnMasteryStatChanged(func(sim *core.Simulation, oldMastery, newMastery float64) {
+		war.PseudoStats.EnragedEffectMultiplier = war.GetMasteryBonusMultiplier()
+	})
+}
```

Comment:
If this mastery affects specific spells then there is no reason to keep this EnragedEffectMultiplier as a cached stat. The spell can just calculate the effect when they are being executed and you will get rid of a very warrior specific stat in the pseudoStats which to me is kinda ugly